query-connector/src/app/database-service.ts

```diff
@@ -391,3 +403,51 @@ function logRejectedPromiseReasons<T>(
       return r.reason;
     });
 }
+
+/**
+ * Function that orchestrates query insertion for the query building flow
+ * @param input - Values of the shape UserQueryInput needed for query insertion
+ * @returns - Success or failure status, with associated error message for frontend
+ */
+export async function insertUserDefinedQuery(input: UserQueryInput) {
```

Comment:
All queries will be user defined. I think we can just `insertQuery`.

Reply:
➕